package/base-files/files/bin/board_detect

```diff
@@ -5,8 +5,8 @@ CFG=$1
 [ -n "$CFG" ] || CFG=/etc/board.json
 
 [ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
+	for a in /etc/board.d/*; do
```

Comment:
I would be awfully tempted to substitute `find /etc/board.d -maxdepth 1 -print | sort | while read a; do` as this avoids having the _shell_ scan the directory.

Reply:
Which would not yield equivalent results though, as it returns `/etc/board.d` at depth 0 as well, so it should probably be `-mindepth 1 -maxdepth 1`. 

While generally preferring `find` for finding operations, the line gets pretty unhandy here and would also depend on different flavors of `find` and `sort` (BusyBox, findutils, coreutils)  installed on the target. (though they all should behave the same in this case). I'd therefore _slightly_ prefer the shorter glob version here.